### DISCUSSION TITLE
Clock 2.0

### DISCUSSION THREAD

**Discussion by BlackyHawky** [BlackyHawky/Clock#23] [Announcements]
**Fixes**
- Fix alarm triggering for Android 14;
- Fix stopwatch and timer notifications;
- Fix force close for bedtime when no alarm is set or when the device is rotated;
- Fix some malfunctions for the Bedtime mode;
- Fix screensaver launch for Android 13+;
- Fix display and update of digital widget for Android 12+;
- Fix setting alarm volume in settings;
- Fix background color of bottom navigation bar (tablets are the most affected);

**Improvements**
- Implement material design (switches, time picker, rounded icons, card background,...);
- Add dynamic colors;
- Add AMOLED mode for dark theme;
- Add new permission management;
- Add monochrome icon;
- Add collapsing toolbar;
- Add Material You analog and digital widgets;
- Add "Bedtime" mode (by @Nilsu11);
- Replace icons with more rounded ones;
- When the application is launched for the first time, a presentation of the main features as well as the possibility of managing permissions is included;
- Add verification that essential permissions are granted at app launch to ensure the app is working properly;
- Allow debug version to be installed along with release version;
- Set default ringtone for alarms in settings;
- Add screensaver display options;
- Activate screensaver with a long press on the clock;
- Reduce screensaver animation to 30 seconds;
- Menus are displayed by icons;
- Modernize the appearance of settings;
- Add "About" section in settings;
- Animate the bedtime icon;
- Displays keyboard automatically when editing an alarm;
- Add a delete icon and the text "Delete" when swiping to delete an alarm;
- Add a button to duplicate an alarm (except for the bedtime alarm, which is useless);
- The vibrator is not activated when an alarm is created and on alarms initially created;
- Increase vibrator duration when activating an alarm;
- Allow display of analog clock in the alarm trigger view;
- Update Fragment related code to use AndroidX;
- Update license headers;
- Delete unnecessary files and code;

<hr /><em>This discussion was created from the release <a href='https://github.com/BlackyHawky/Clock/releases/tag/2.0'>Clock 2.0</a>.</em>

**Comment by qw123wh** [comment]
Perfect, thank you. 

I noticed that you removed do not disturb for bedtime, you did well; actually not even the google clock activates do not disturb. At this point you could also remove the do not disturb permission also because that permission was only needed for bedtime

Add topic in the about section, you can use the same ones I put in the old reposity.
Without the topics, this repository is impossible to find for a user searching for an Android clock app on github.

**Reply by BlackyHawky** [reply]
> At this point you could also remove the do not disturb permission also because that permission was only needed for bedtime

As I didn't have many devices to test, I preferred to leave this permission; moreover, I realized that it is present and necessary in other alarm applications. Surely this can be removed in a future release.

> Add topic in the about section...

Done. Thanks for the tip. 😉

**Reply by Nilsu11** [reply]
Since I added the permission I also can confirm that it only was used in the bedtime service, so it can be removed safely. If you want notifications to ignore DND you would have to regulate this at the notification channel mangement in NotificationUtils which I haven't touched. Also google clock doesn't require it, which this clock app is closely related to. Since you removed all bedtime mode features you could also go ahead and remove it.

**Reply by BlackyHawky** [reply]
I hadn't realized that this authorization was added at the same time as the bedtime mode. So I'm going to remove it without worry.
(If I'd known, I'd have released it right after this change... Too bad.)

**Comment by BlackyHawky** [comment]
I'm not sure that the update will be done on F-Droid; indeed, if you look [here](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.best.deskclock.yml), you can see that the old repository is still referenced (but the version has been updated correctly).
Perhaps I need to open a ticket?

**Reply by odmfl** [reply]
I also noticed that it doesn't update on Android.  I don't know if we have to wait a few days, in my opinion it's better to open a ticket.

In the topics also add "android" "foss" "fdroid", these are keywords in addition to "clock" that the user writes in the github search

**Reply by BlackyHawky** [reply]
What do you mean by "_I also noticed that it doesn't update on Android._" ?

>In the topics also add "android" "foss" "fdroid"...

Done 😉

**Reply by qw123wh** [reply]
sorry, meant that the app had not been updated on Fdroid 😅😅

**Reply by odmfl** [reply]
Yes I reversed the words "Android" and "Fdroid" 😂😂

**Reply by BlackyHawky** [reply]
That's what I thought. 😅

From what I've read on Gitlab, it can take several days for the version to be compiled on F-Droid.
So I'll see next weekend whether I'll open a ticket or not.

**Reply by BlackyHawky** [reply]
@IzzySoft: I've seen you intervene in the past to make this application available on F-Droid  so I'd like to ask for your help and explanations, please.

I've made a few modifications, published a release version on GitHub available [here](https://github.com/BlackyHawky/Clock/releases/tag/2.1)) and it would be great if the application could also be updated on F-Droid.

I can see that the _com.best.deskclock.yml_ file (available [here](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.best.deskclock.yml)) is updated, but only partially.
Indeed, the `SourceCode:`, `IssueTracker:` and `Repo:` fields are not updated and I have the impression that this will prevent the application from being updated on F-Droid.
(FYI: these fields refer to the old repository that @qw123wh transferred to me a few days ago.)

Can you tell me if I'm right or not? If so, do you know how to update these fields? Maybe it's automatic after a few days?

Thank you in advance.

Edit: I've opened an issue on GitLab ([here](https://gitlab.com/fdroid/fdroiddata/-/issues/3259)) in case this is necessary.

**Reply by IzzySoft** [reply]
I'm currently [not active at F-Droid](https://gitlab.com/fdroid/admin/-/issues/447), so it won't be me to intervene there this time. Also with so many apps I've worked on, I don't remember the reasons for each of them. I'd suggest you to report that there. With the original repo transferred to you, there should be no conflicts when it comes to "take over the packageName" etc.

for updating the mentioned fields, it will need a merge request at fdroiddata.

**Reply by BlackyHawky** [reply]
I didn't know there were any disputes within the F-Droid team...

Finally, we've already requested the changes in the metadata file; these changes have been validated, so let's wait and see if the update is done correctly.

@IzzySoft Thank you for taking the time to reply.

**Reply by IzzySoft** [reply]
> I didn't know there were any disputes within the F-Droid team...

Disputes? Eh. Just a few (core) contributors (including their RB specialist) left, so what `</s>`

Nevertheless, best luck and good progress for you!

**Comment by qw123wh** [comment]
@BlackyHawky I tried to create a PR where I change the route maybe this can help. Let's wait a few more days.


https://gitlab.com/fdroid/fdroiddata/-/merge_requests/14985

**Reply by BlackyHawky** [reply]
Thanks.
I'll be watching closely as I tried an experiment today with updates.
It's hard to explain clearly, but I want to see if tomorrow, the last commit 125bc7a is correctly taken into account in the [metadata](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.best.deskclock.yml) automatically.

If this is not the case, you'll need to update the PR on GitLab by replacing line 29 (the `commit` field) of the metadata file. So this is what it will look like:

```yml
  - versionName: '2.1'
    versionCode: 2002
    commit: 125bc7ad017ac4169994db85274784953d94b8b7
    subdir: app
    gradle:
      - yes

```

BTW, in relation to my experience, if you already own version 2.1 of the application on or before 07 may 2024, please download and reinstall the apk.
A last-minute patch has been added without (exceptionally) modifying the version (commit ae897bd).
Sorry for the inconvenience.

**Reply by qw123wh** [reply]
My PR has been merged, I need to modify the commit, but I can't, it gives me an error.

Now try modifying it yourself on Gitlab or we'll wait a few days. In this period, however, we avoid updates and various experiments. I don't know if the solution is to release a 2.3 version but I would wait. 

We are driving fdroid crazy with all these changes 😂😂😂

**Reply by odmfl** [reply]
@BlackyHawky @qw123wh stop everyone, I did the PR for modifying the commit, let's wait for the merge and wait for the release, then we can proceed to release new versions

https://gitlab.com/fdroid/fdroiddata/-/merge_requests/14993

**Reply by qw123wh** [reply]
@odmfl Ok, your PR has been merged. We just have to wait a few days. If the app does not update on fdroid, you will need to open a ticket

**Reply by BlackyHawky** [reply]
Thank you both for doing it on GitLab. I don't have time to log in at the moment.

Anyway, I don't think it's necessary to create a 2.3 version because thanks to you all the links are now correct. The next F-Droid server updates will take place soon, so I'll wait.

Obviously, I won't be doing any new experiments until then, which by the way confirmed that my manipulation doesn't work between GitHub and the data on GitLab.

No changes will be made to the application until it's available on F-Droid.

As a last resort, I'll open a ticket on GitLab.

**Reply by BlackyHawky** [reply]
Today, version 2.0 has been published on F-Droid.

So we're well on the way to releasing version 2.1 as well.

However, it's too bad that the old screenshots are still present on F-Droid...
Perhaps by the time version 2.1 is released, these captures will have disappeared.

**Reply by qw123wh** [reply]
Great, let's wait for version 2.1

I noticed that the new screenshots have been added, but the old ones are also there. Maybe with version 2.1 it will fix itself

**Reply by BlackyHawky** [reply]
> I noticed that the new screenshots have been added, but the old ones are also there. Maybe with version 2.1 it will fix itself.

Normally yes, because if you look at the issue tracker, it's always the old deposit that's taken into account. So let's be optimistic 😉

**Reply by IzzySoft** [reply]
The screenshots won't fix themselves. It's a long-standing bug with fdroidserver. A work-around is to replace the unwanted ones by 1x1 px images, and once those have overwritten the old screenshots remove them again from fastlane.

**Reply by BlackyHawky** [reply]
@IzzySoft Thank you so much for your tip!
Can you confirm that it's not necessary to update the application version?

Edit: commit 7e5df63 done

**Reply by IzzySoft** [reply]
I cannot confirm, sorry. At F-Droid.org, the changes will be pulled with the corresponding release, and all data used (code and fastlane) is pulled from the corresponding release tag. I'm not aware of any exceptions being made to update from fastlane between releases (at F-Droid.org that is; at IzzyOnDroid that's possible and has been done when needed – but then, at IzzyOnDroid those screenshots could have been cleaned up completely without using "dummies").

**Reply by BlackyHawky** [reply]
Don't be sorry. It's already great that you gave me the tip to be able to delete those screenshots.
I'll do what's necessary when version 2.1 is released.
Thanks again.

**Reply by BlackyHawky** [reply]
@IzzySoft: Indeed, you were right: I've just seen on the Droid-ify app that version 2.1 has been released and the screenshots haven't been removed.
All I have to do now is apply your tip.
In any case, thank you so much for giving me this famous tip!

**Reply by IzzySoft** [reply]
Fingers crossed! I've recommended this approach a few times until now, and it brought the hoped-for result – so chances are good.

**Reply by BlackyHawky** [reply]
> The screenshots won't fix themselves. It's a long-standing bug with fdroidserver. A work-around is to replace the unwanted ones by 1x1 px images, and once those have overwritten the old screenshots remove them again from fastlane.

@IzzySoft 
- Screenshots replaced for version 2.2 with commit 7e5df635;
- Screenshots removed for version 2.3 with commit b4a9ae4f;

but screenshots are still available in F-Droid 😭😭

**Reply by IzzySoft** [reply]
As outlined: long-standing bug with F-Droid which I doubt they'll fix anytime soon. Just add 1x1px pictures to replace the old ones so they become invisible. Once the next release was pulled and the 1x1px replacements reached F-Droid, you can remove them again from your fastlane (which will automatically remove them from IzzyOnDroid with the release after that).

Pointer: Never rename screenshots if your app is listed at F-Droid.org, their updater does not handle that.

**Reply by IzzySoft** [reply]
@BlackyHawky as those 1x1 px screenshots now served their purpose (they've reached F-Droid), maybe it's time to remove them here again to not "pollute" other places using the structure (e.g. IzzyOnDroid, where inclusion was requested and your app added meanwhile, too)? Thanks in advance!

**Reply by BlackyHawky** [reply]
@IzzySoft 
The last 1x1 px screenshots added have been removed for version 2.10.1

**Reply by IzzySoft** [reply]
Thanks! I've now updated the config here to pull screenshots along with the other fastlane data on updates.
